### PR TITLE
AC-7848 P0: Directory pages returning error for users

### PR DESCRIPTION
**Pull request by geofrocker**
### Changes Introduced in [AC-7848](https://masschallenge.atlassian.net/browse/AC-7848):
- Removed program status check
### Testing:
- Check out this branch
- Masquerade as a finalist
- The people directory and look up a finalist whose program is no longer active
- Notice that you can view this user's profile

### CODE REVIEW

File: web/impact/impact/tests/test_graphql.py [hunk 1]
```diff
@@ -604,23 +604,6 @@ def test_query_for_user_without_confirmed_mentor_program_families(self):
             self.assertEqual(expert_profile["confirmedMentorProgramFamilies"],
                              [])
 
-    def test_user_cannot_view_profile_with_non_current_allowed_roles(self):
```
Comment (jonkiparsky) [inline web/impact/impact/tests/test_graphql.py:607]: Rather than deleting this test, let's change it to reflect the behavior change

**Review by freshprincekla — APPROVED** [review]
QA check out:
- masqueraded as a finalist and went to another finalist whose program was no longer active and I was able to see that their profile.

**Review by jonkiparsky — CHANGES_REQUESTED** [review]
The code change looks okay. I'm trying to understand the overall requirements of the directories, since this change reverts a change made recently by @iankigen and I'd like to make sure that whatever he was fixing is not being broken here.

File: web/impact/impact/tests/test_graphql.py (hunk 1 above)
Comment (geofrocker) [reply web/impact/impact/tests/test_graphql.py:607]: modified the test

**Review by jonkiparsky — APPROVED** [review]

**Pull request merged**